exercises/yacht/Example.cs

```diff
@@ -0,0 +1,97 @@
+    ///
+    ///    Scores in Yacht
+    ///    Category    Score Example
+    ///    Ones            1 × number of ones      1 1 1 4 5 scores 3
+    ///    Twos            2 × number of twos      2 2 3 4 5 scores 4
+    ///    Threes          3 × number of threes    3 3 3 3 3 scores 15
+    ///    Fours           4 × number of fours     1 2 3 3 5 scores 0
+    ///    Fives           5 × number of fives     5 1 5 2 5 scores 15
+    ///    Sixes           6 × number of sixes     2 3 4 5 6 scores 6
+    ///    Full House      Total of the dice       3 3 3 5 5 scores 19
+    ///    Four of a Kind  Total of the four dice  4 4 4 4 6 scores 16
+    ///    Little Straight 30 points               1 2 3 4 5 scores 30 
+    ///    Big Straight    30 points               2 3 4 5 6 scores 30
+    ///    Choice Sum of the dice         2 3 3 4 6 scores 18
+    ///    Yacht           50 points               4 4 4 4 4 scores 50
+    ///    If the dice do not satisfy the requirements of a category, the score is zero.If, for example,
+    ///    Four Of A Kind is entered in the Yacht category, zero points are scored. A Yacht scores zero 
+    ///    if entered in the Full House category.
+    ///
+    ///    Task
+    ///    Given a list of values for five dice and a category, your solution should return 
+    ///    the score of the dice for that category. If the dice do not satisfy the requirements
+    ///    of the category your solution should return 0. You can assume that five values will always be presented,
+    ///    and the value of each will be between one and six inclusively. You should not assume that the dice are ordered.
+    /// </summary>
+    /// <param name="diceSet">Dice Array</param>
+    /// <param name="category">Name of category</param>
+    /// <returns></returns>
+    public static int Score(int[] diceSet, string category)
+    {
+        switch (category.ToLower())
+        {
+            //Ones: Any combination The sum of dice with the number 1.
+            case "ones": return diceSet.Where(x => x == 1).Sum();
+            //Twos: Any combination The sum of dice with the number 2. 
+            case "twos": return diceSet.Where(x => x == 2).Sum();
+            //Threes: Any combination The sum of dice with the number 3.
+            case "threes": return diceSet.Where(x => x == 3).Sum();
+            //Fours: Any combination The sum of dice with the number 4.
+            case "fours": return diceSet.Where(x => x == 4).Sum();
+            //Fives: Any combination The sum of dice with the number 5.
+            case "fives": return diceSet.Where(x => x == 5).Sum();
+            //Sixes: Any combination The sum of dice with the number 6.
+            case "sixes": return diceSet.Where(x => x == 6).Sum();
+
+            //Full House:  Three of one number and two of another.  Sum of all dice.
+            case "full house":
+                {
+                    var testHashSet = new HashSet<int>();
```

Comment:
I think you can simplify this logic by doing the following:

`diceSet.Distinct().Count() == 2 ? diceSet.Sum() : 0;`